src/Mainloop.cu

```diff
@@ -56,6 +56,14 @@ template <class T> void MainLoop(Param &XParam, Forcing<float> XForcing, Model<T
 		resetmeanmax(XParam, XLoop, XModel, XModel_g);
 
 		printstatus(XLoop.totaltime, XLoop.dt);
+
+		// Stop calculation if crashed
+		if (XLoop.dt < XParam.dtmin)
+		{
+			XLoop.totaltime = XParam.endtime + 1;
```

Comment:
would make more sense to say :
XParam.endtime = XLoop.totaltime
with the same outcome.
i.e. when looking at the model status after the crash it will look like the model reached the expected endtime when in reality it stopped earlier.

Reply:
Good comments, I will make these changes